eip-0020.md

```diff
@@ -0,0 +1,246 @@
+[EIP-0019](eip-0019.md).
+
+Below we describe the data formats specified by this EIP and refer to the
+formats defined in EIP-0019.
+
+Wallet apps should be able to initiate ErgoPay both by using URI schemes 
+(clickable links) or QR codes.
+
+### Option 1: dApp providing URL request to fetch information from
+
+`ergopay://<URL>`
+
+URL is provided without the https prefix. http communication is not allowed except for IP addresses 
+(in order to test within a local network).
+
+The provided URL can contain a `#P2PK_ADDRESS#` placeholder. The wallet application will replace this 
+placeholder with an actual P2PK address.
+
+Examples:
+* `ergopay://sigmausd.io/signingRequest/2001-16b8-66c4-b800-6e52-8ce4` will make the wallet app request
+`https://sigmausd.io/signingRequest/2001-16b8-66c4-b800-6e52-8ce4`
+* `ergopay://192.168.0.1/html` will make the wallet app request
+`http://192.168.0.1/html`
+* `ergopay://auctionhouse.org/bid/#P2PK_ADDRESS#` will make the wallet app request `ergopay://auctionhouse.org/bid/3Wx7Z8FywaL4ofunDCV1NaTJX5CpjTubMrjpCkEhnNBAgJLGfRcD`
+
+#### Response: ErgoPaySigningRequest
+
+The wallet application should request URL and obtain the following data (json format)
+
+```
+ErgoPaySigningRequest:
+  - transaction: ReducedTransaction (optional*)
+  - p2pkaddress: String (optional)
+  - message: String (optional*)
+  - messageSeverity: String (optional) "INFORMATION", "WARNING", "ERROR"
+  - replyToUrl: String (optional)
+```
+
+Either a **transaction** or a **message** must be provided, otherwise the request is invalid.
+
+The wallet application should show the **message** and display the **messageSeverity** in a suitable 
+way, if provided.
+
+If **transaction** details are obtained, a payment screen opens for the user to confirm 
+signing the transaction. Transaction details CAN BE OPTIONALLY shown on the payment
+screen of the wallet application. 
+
+If **p2pkaddress** is provided by the dApp, the wallet can preselect the key the user
+needs to sign the transaction.
+
+After signing is performed and the `SignedTransaction` data is obtained, the 
+wallet can POST the following data to the dApp using **replyToUrl** from the
+signing request (json format) if it is provided. 
+
+```
+ErgoPayTransactionId:
+  - txId: String
+```
+
+dApps should not rely on this request to be made. It could happen that the transaction was 
+submitted to a node, but the reply couldn't be executed. dApps know the transaction id and
+should monitor the mempool and blockchain on their own.
+
+In case no **transaction** was provided, the wallet app displays the **message** that should inform the
+user about further steps needed.
+
+
+### Option 2: dApp providing URI scheme containing reduced transaction
+
+When dapp does not need to use an extra request, the `ReducedTransaction` could 
+also be encoded in the QR code or link:
+
+`ergopay:<ReducedTransaction, base 64 url safe encoded>`
```

Comment:
maybe a note or a link to other docs that "url safe" denotes a replacement of `'+' = '-'` and `'/' = '_'`

Reply:
It's a standard encoding format, implemented in most programming languages. So I think we don't need a further definition here.